wikihow2zim/scraper.py

```diff
@@ -76,12 +76,161 @@ def add_assets(self):
+            soup.find("div", {"id": "intro"}).find("div", {"id": "mf-section-0"}).text
+        )
+
+        # Get the methods/parts
+        methods = []
+
+        for node in soup.find_all("div", {"class": "steps"}):
+
+            method = {}
+
+            # Get method title
+            method["title"] = node.find("span", {"class": "mw-headline"}).text
+
+            # Get method sections:
+            method["steps"] = []
+            steps_list = node.find("div", {"class": "section_text"}).select(
+                "li[id*=step-id]"
+            )
+
+            # Get the title, text and images for all the steps in the given method/part
+            for s in steps_list:
+
+                step = {}
+                image = s.find("a", {"class": "image"})
+                if image is not None:
+                    step["image"] = image["href"]
+                else:
+                    step["image"] = None
+
+                step["title"] = str(
+                    s.find("div", {"class": "step"}).find("b", {"class": "whb"}).text
+                )
+
+                text = s.find("div", {"class": "step"})
+                if text.find("script"):
+                    text.find("script").decompose()
+                if text.find("b", {"class": "whb"}):
+                    text.find("b", {"class": "whb"}).decompose()
+                sups = text.find_all("sup")
+                for ss in sups:
+                    ss.decompose()
+                step["text"] = text
+
+                method["steps"].append(step)
+
+            methods.append(method)
+
+        sys.exit()
+
+        # article = self.env.get_template("article.html")
+        # self.creator.add_item_for(
+        #     path="category/" + article_title,
+        #     title="article_title",
+        #     content=article.render(title=article_title, description=description),
+        #     mimetype="text/html",
+        # )
+
+        # Add images:
+        # url = (
+        #     "https://www.wikihow.com/extensions/wikihow/mobile"
+        #     "/images/wikihow_logo_230.png"
+        # )
+        # self.creator.add_item(
+        #     URLItem(
+        #         url=url,
+        #         path="assets/static/wikihow_logo.png",
+        #         mimetype="image/png",
+        #     )
+        # )
+
+    def walk_subcategories(self, main_wiki_url, cat_url, recursion_depth):
+        try:
+            response = requests.get(cat_url)
+        except Exception as exc:
+            logger.critical(f"Unable to retrieve homepage at {cat_url}: {exc}")
+            logger.exception(exc)
+
+        soup = bs4.BeautifulSoup(response.text, "html.parser")
+        article_divs = soup.find_all("div", {"class": "responsive_thumb"})
+        for article_div in article_divs:
+            article_url = article_div.find("a")["href"]
+            if not article_url in self.articles_list:
+                self.articles_list[article_url] = [str(recursion_depth) + " " + cat_url]
```

Comment:
what's the purpose of this ? Debug print only?

Reply:
Yes, I used this to see how deep the recursion goes, and also how many places an article is accessed from. I can delete these lines when I'm done.